ticker-tracker/src/main/scala/com/markit/tickertracker/Tracker.scala

```diff
@@ -0,0 +1,100 @@
+package com.markit.tickertracker
+
+import java.io.{ByteArrayInputStream, InputStreamReader}
+import java.time.{LocalDate, Period, ZoneOffset}
+import java.util.logging.Logger
+
+import cats.data.ValidatedNel
+import com.github.tototoshi.csv.CSVReader
+import com.outworkers.util.catsparsers._
+import com.twitter.concurrent.AsyncStream
+import com.twitter.finagle.http.Status
+import com.twitter.finagle.{Http, Service, http}
+import com.twitter.io.{Buf, Reader}
+import com.twitter.util.Future
+
+class TickerSymbol(val value: String)
+
+object TickerSymbol {
+  case object GOOG extends TickerSymbol("GOOG")
+}
+
+trait Tracker {
+
+  def logger: Logger = Logger.getLogger("tracker")
+
+  private[this] def host: String = "real-chart.finance.yahoo.com:80"
+
+  def client: Service[http.Request, http.Response] = Http.client.withStreaming(enabled = true).newService(host)
+
+  protected[tickertracker] def parseResponse(
+    httpResponse: http.Response
+  ): Future[Iterator[ValidatedNel[String, DailyValue]]] = httpResponse match {
+    case response if response.status != Status.Ok => Future.exception(new Exception(s"Status code ${response.status}"))
+    case resp => resp.reader.read(Int.MaxValue) map {
```

Comment:
If you wanted to do this "properly" you could use a `PipedOutputStream` being fed the bytes from the reader on a loop and hook up a `PipedInputStream` to it. This could be passed to the csv reader, giving you "true" streaming. You'd of course need to be cute about how to return the completed streaming computation, but I'm sure you'd figure something out.